src/main/java/org/opensearch/flowframework/common/DefaultUseCases.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.common;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Enum encapsulating the different default use cases and templates we have stored
+ */
+public enum DefaultUseCases {
+
+    /** defaults file and substitution ready template for OpenAI embedding model */
+    OPEN_AI_EMBEDDING_MODEL_DEPLOY(
+        "open_ai_embedding_model_deploy",
+        "defaults/open-ai-embedding-defaults.json",
+        "substitutionTemplates/deploy-remote-model-template.json"
+    ),
+    /** defaults file and substitution ready template for cohere embedding model */
+    COHERE_EMBEDDING_MODEL_DEPLOY(
+        "cohere-embedding_model_deploy",
+        "defaults/cohere-embedding-defaults.json",
+        "substitutionTemplates/deploy-remote-model-template-extra-params.json"
+    ),
+    /** defaults file and substitution ready template for local neural sparse model and ingest pipeline*/
+    LOCAL_NEURAL_SPARSE_SEARCH(
+        "local_neural_sparse_search",
+        "defaults/local-sparse-search-defaults.json",
+        "substitutionTemplates/neural-sparse-local-template.json"
+    );
```

Comment:
We should provide an easy way for users to add their own `default` files and `substitutionTemplate`. Probably a json/yaml file which contains the mapping of the default file and substitution template. Having an enum would require them to add here.

Reply:
Are you suggesting not using the enum? Enum at least allows in compile time guarantee that users have both a use case name, a default file and a template file all registered together for that use case. relaying on a json to ensure we have both means more runtime errors.

I do agree with you though that it means they have to add things in multiple places, this however isn't final version of this and I was hoping in 2.14/15 to have an API for user to add more usecases

Reply:
Can you create an issue to address this? This can be make more cleaner. Having enum would ask user to add code here.

Reply:
yeah this is a matter of what is easier for user to add more too and what can be more safe in terms of runtime error. I am still not against having some system index users can just hit API to add more at least on there cluster so don't need to touch code

Reply:
created issue https://github.com/opensearch-project/flow-framework/issues/584

Reply:
> We should provide an easy way for users to add their own default files and substitutionTemplate

Definitely for the next version.  I think we need discussion on where to store those (all in same template or separately) as either way is a one-way decision with change to system indices.